src/transaction/atomic_replication_helper.hpp

```diff
@@ -79,13 +82,15 @@ namespace cublog
 			       const log_rv_redo_rec_info<T> &record_info);
 	  int fix_page (THREAD_ENTRY *thread_p);
 	  void unfix_page (THREAD_ENTRY *thread_p);
+	  LOG_LSA get_lsa () const;
 
-	  VPID m_vpid;
 	private:
-	  log_lsa m_record_lsa;
+	  const VPID m_vpid;
```

Comment:
I think this implementation is nor generic enough nor, as a consequence, correct.
It does not cover the situation when, during an atomic replication sequence, there is more than one log record that modifies the same page.
The code needs to be refactored:
- a structure needs to keep a shared pool of `pgptr`'s
- an atomic replication unit must only keep a "weak reference" to a page ptr (eg, more than one atomic replication unit can reference the same `pgptr` kept in this separate structure
- when a new replication unit is about to be added, it will check, based on `vpid` whether the page has already been fixed and, if yes, take the `pgptr` from the shared structure
- if not, will fix the `pgptr` and add it to the shared structure/collection for use by other atomic replication units
I will add a separate issue for this to be implemented separately and then integrated in this PR.